platform/spire/src/setup.py

```diff
@@ -226,7 +226,13 @@ def modify_dns_bootstrap(ops: Operations, is_install: bool) -> None:
 
 def dns_bootstrap_lines() -> str:
     config = configuration.get_config()
-    return "".join("%s\t%s # AUTO-HOMEWORLD-BOOTSTRAP\n" % (ip, hostname) for hostname, ip in config.dns_bootstrap.items())
+    dns_hosts = list(config.dns_bootstrap.items())
+    for node in config.nodes:
+        if node.kind == "supervisor":
+            dns_hosts += [("homeworld.private", node.ip)]
```

Comment:
This won't work correctly if we have more than one supervisor. That's fine, since we already don't support having more than one supervisor, but can we make the code fail fast in this situation, instead of silently generating a bad `/etc/hosts` file?

One way to do this would be to just use the `config.supervisor` property, which already fails-fast during generation when more than one supervisor is specified; that way, if we ever support more than one, we'll notice that this code expects just one while repairing uses of that field.